distributed/deploy/ssh.py

```diff
@@ -86,25 +86,37 @@ def __init__(
         import asyncssh  # import now to avoid adding to module startup time
 
         self.connection = await asyncssh.connect(self.address, **self.connect_options)
-        self.proc = await self.connection.create_process(
-            " ".join(
-                [
-                    'DASK_INTERNAL_INHERIT_CONFIG="%s"'
-                    % serialize_for_cli(dask.config.global_config),
-                    sys.executable,
-                    "-m",
-                    self.worker_module,
-                    self.scheduler,
-                    "--name",
-                    str(self.name),
-                ]
-                + cli_keywords(self.kwargs, cls=_Worker, cmd=self.worker_module)
-            )
+        cmd = " ".join(
+            [
+                'env DASK_INTERNAL_INHERIT_CONFIG="%s"'
+                % serialize_for_cli(dask.config.global_config),
+                sys.executable,
+                "-m",
+                self.worker_module,
+                self.scheduler,
+                "--name",
+                str(self.name),
+            ]
+            + cli_keywords(self.kwargs, cls=_Worker, cmd=self.worker_module)
         )
 
+        self.proc = await self.connection.create_process(cmd)
+
         # We watch stderr in order to get the address, then we return
+        retry_flag = True
         while True:
             line = await self.proc.stderr.readline()
+            # retry without env command
+            if not line.strip() and retry_flag:
+                retry_flag = False
+                logger.debug("Retry creating process without env command")
+                self.proc = await self.connection.create_process(
+                    cmd.replace(
+                        "env DASK_INTERNAL_INHERIT_CONFIG=",
+                        "DASK_INTERNAL_INHERIT_CONFIG=",
```

Comment:
Since the original suggestion was about windows' lack of `env`, does this do the right thing?

Reply:
Not really that familiar with Windows environment ... should I add another retry to run `set DASK_INTERNAL_CONFIG=...`? Can I determine the OS type for the host using `asyncssh` after establishing the connection?

Reply:
Ah, actually good point, probably an ssh host isn't windows. 
I would just catch the error and repot it the first time in that case, assume that the ssh host *must* have env to work (and not force to retry due to any other error).